presto-main/src/main/resources/webapp/src/components/ResourceGroupView.jsx

```diff
@@ -0,0 +1,349 @@
+        id: [],
+        state: '',
+        schedulingPolicy: '',
+        schedulingWeight: '',
+        softMemoryLimit: '',
+        softConcurrencyLimit: 0,
+        hardConcurrencyLimit: 0,
+        maxQueuedQueries: 0,
+        numEligibleSubGroups: 0,
+        workersPerQueryLimit: 0,
+        subGroups: [],
+        runningQueries: [],
+        rootGroups: [],
+
+        numRunningQueries: 0,
+        numQueuedQueries: 0,
+        memoryUsage: 0,
+        numAggregatedQueuedQueries: 0,
+        numAggregatedRunningQueries: 0,
+
+
+        activeWorkers: 1,
+        rowsPerSec: 0.00,
+        runnableDrivers: 0,
+        bytesPerSec: 0,
+        blockedQueries: 0,
+        reservedMemory: 0,
+        workerParallelism: 0,
+        showDoc: true,
+        showResource: false,
+        error: null,
+    });
+
+    function fetchData() {
+        if (group === null) {
+            // fetch root groups
+            fetch('/v1/resourceGroupState/')
+            .then(response => response.json())
+            .then((resources) => {
+                setValues({...values, showDoc: true, showResource: false,
+                        rootGroups: resources, error: {message: 'No resource group data. Resource groups data will be populated once a SQL query is handled by a coordinator.'}});
+            });
+            return;
+        }
+        fetch('/v1/resourceGroupState/' + group.replace('.', '/'))
+        .then(response => response.json())
+        .then((resources) => {
+            // console.log(`${JSON.stringify(resources, null, '  ')}`);
+            dataSet.current = {
+                    numRunningQueries: addToHistory(resources.numRunningQueries, dataSet.current.numRunningQueries),
+                    numQueuedQueries: addToHistory(resources.numQueuedQueries, dataSet.current.numQueuedQueries),
+                    numAggregatedRunningQueries: addToHistory(resources.numAggregatedRunningQueries, dataSet.current.numAggregatedRunningQueries),
+                    numAggregatedQueuedQueries: addToHistory(resources.numAggregatedQueuedQueries, dataSet.current.numAggregatedQueuedQueries),
+                    memoryUsage: addToHistory(parseDataSize(resources.memoryUsage), dataSet.current.memoryUsage),
+            };
+
+            setValues({...values,
+                id: [...resources.id],
+                state: resources.state,
+                schedulingPolicy: resources.schedulingPolicy,
+                schedulingWeight: resources.schedulingWeight,
+                softMemoryLimit: resources.softMemoryLimit,
+                softConcurrencyLimit: resources.softConcurrencyLimit,
+                hardConcurrencyLimit: resources.hardConcurrencyLimit,
+                maxQueuedQueries: resources.maxQueuedQueries,
+                numEligibleSubGroups: resources.numEligibleSubGroups,
+                workersPerQueryLimit: resources.workersPerQueryLimit,
+                subGroups: resources.subGroups.map(subgrp => subgrp.id.join('.')),
```

Comment:
It seems to me it is always good practice to stick with `camelCase`, so `subgrp` to `subGrp` or even better `subGroup`

Reply:
let me change it to `subGroup`